internal/mode/static/metrics/collectors/nginx.go

```diff
@@ -40,8 +43,34 @@ func NewNginxPlusMetricsCollector(constLabels map[string]string) (prometheus.Col
 	if err != nil {
 		return nil, err
 	}
-	variableLabelNames := nginxCollector.VariableLabelNames{}
-	return nginxCollector.NewNginxPlusCollector(plusClient, metrics.Namespace, variableLabelNames, constLabels), nil
+
+	promLogger, err := newPromLogger()
+	if err != nil {
+		return nil, err
+	}
+
+	collector := nginxCollector.NewNginxPlusCollector(
+		plusClient,
+		metrics.Namespace,
+		nginxCollector.VariableLabelNames{},
+		constLabels,
+		promLogger,
+	)
+
+	return collector, nil
+}
+
+// newPromLogger creates a Prometheus logger that implements to go-kit log.Logger that the prometheus exporter requires.
+func newPromLogger() (log.Logger, error) {
+	logFormat := &promlog.AllowedFormat{}
+
+	if err := logFormat.Set("json"); err != nil {
+		return nil, err
+	}
+
+	logConfig := &promlog.Config{Format: logFormat}
```

Comment:
my only concern is that we don't try to match the logging levels of NGF to logging levels of prometheus logger. the only thing that we match - format (json).

do we want to control/configure the log level?
looking at the library, it supports it

if we don't do that, would it make sense to add a comment that we intentionally don't do that?

```
// NewDynamic returns a new leveled logger. Each logged line will be annotated
// with a timestamp. The output always goes to stderr. Some properties can be
// changed, like the level.
func NewDynamic(config *Config) *logger {
```

Reply:
Ideally we'd be able to inject our logger into the prometheus collector, but the logging interface we use is different from the one prometheus exporter accepts.

If we do want to control the level of the prometheus logger, here's one way we can do it: https://github.com/nginxinc/nginx-gateway-fabric/pull/1347/commits/ef0508ae7859dd0d268c95991605c9a6291d9c50.

What do you think?

Reply:
maybe we can refactor prometheus exporter so that it accepts some narrow logging interface that it can define and we can implement two adapters for it - one for logr (so that we can pass our logger) and one for promlog in the exporter project.

This project will also need to use the new interface https://github.com/nginxinc/kubernetes-ingress/pull/4769/files

I wonder what other people think.

At the same time, I left comments to the approach you took in https://github.com/nginxinc/nginx-gateway-fabric/commit/ef0508ae7859dd0d268c95991605c9a6291d9c50

Reply:
> maybe we can refactor prometheus exporter so that it accepts some narrow logging interface that it can define and we can implement two adapters for it - one for logr (so that we can pass our logger) and one for promlog in the exporter project.

I think that would be a breaking change for the exporter library, right? Not sure that will fly.